torch/onnx/_internal/exporter.py

```diff
@@ -522,12 +451,39 @@ def export(self) -> ExportOutput:
             *self.model_args, **self.model_kwargs
         )
 
-        # Export FX graph to ONNX ModelProto.
-        #
-        # Note that ALL kwargs are folded into constants in graph_module, so we don't pass kwargs
-        # to _export.
-        return self.options.fx_tracer._export_fx_to_onnx(
-            self.options, graph_module, updated_model_args
+        # TODO: Design the passes API
+        graph_module = pre_export_passes(self.options, graph_module, updated_model_args)
+
+        from onnxscript.function_libs.torch_lib import (  # type: ignore[import]
```

Comment:
😢  From this
```python
            onnxscript_graph = passes.export_fx_to_onnxscript(
                diagnostic_context, module, options
            )
```
to this
```python
        onnxscript_graph = onnxscript_graph_building.TorchScriptGraph()

        from torch.onnx._internal.fx import fx_onnx_interpreter

        fx_onnx_interpreter_ctx = fx_onnx_interpreter.FxOnnxInterpreterContext(
            torchscript_tracer=onnxscript_graph_building.TorchScriptTracingEvaluator(
                onnxscript_graph
            ),
            onnxscript_graph=onnxscript_graph,
            fx_graph_module=graph_module,
            fx_name_to_onnxscript_value={},
        )
        fx_interpreter = fx_onnx_interpreter.FxOnnxInterpreter(
            onnxfunction_dispatcher=self.options.onnxfunction_dispatcher,
            diagnostic_context=self.options.diagnostic_context,
            op_level_debug=self.options.op_level_debug,
        )
        fx_interpreter.run(fx_onnx_interpreter_ctx)
```
Let's move initializations into `FxOnnxInterpreter.run`, then in `FxOnnxInterpreter.run` call the staticmethods with those variables explicitly.
Also make `onnxscript_graph` the return value of `run`.
Now the extra lines are purely due to being explicit on `options`, I will refrain from that discussion for now.

```python
        from torch.onnx._internal.fx import fx_onnx_interpreter

        fx_interpreter = fx_onnx_interpreter.FxOnnxInterpreter(
            onnxfunction_dispatcher=self.options.onnxfunction_dispatcher,
            diagnostic_context=self.options.diagnostic_context,
            op_level_debug=self.options.op_level_debug,
        )
        onnxscript_graph = fx_interpreter.run(graph_module)
```

Reply:
That is a good idea. 
Will check which state can be moved inside `run`. Some like `diagnostic_context`, `op_level_debug` and `onnxfunction_dispatcher` can't be avoided.

Reply:
Done, that was an excellent suggestion. Thank you